Adjust ch01s02 architecture (#2)
RedHatQuickCourses/rhods-intro: modules/chapter1/pages/section2.adoc

:navtitle: Architecture

Red{nbsp}Hat OpenShift Data Science (RHODS) is a unified platform for data scientists, developers, and IT ops.
The platform is designed to manage the complete lifecycle of AI/ML projects in the hybrid cloud.

As the following diagram shows, RHODS is deployed as an operator, and includes several components to handle AI/ML development/training, automation pipelines, and model serving:

> image::architecture.svg[width=800px]

Reviewer: This diagram is excellent ! but size can be bigger. I have a hard time making out the small text. Maybe sak Sajith to expand and clean up?

Author: Request created https://redhat.service-now.com/help?id=rh_ticket&table=x_redha_red_hat_tr_x_red_hat_training&sys_id=5bfb746edbf13d54c4db45e813961975

> These architectural components can be organized into the following main concepts:

Reviewer: ```suggestion
The following are the main concepts and terminology as it relates to RHODS:
```

Workbenches::

> A workbench is a containerized, cloud-native data science working environment.

Reviewer: ```suggestion
A workbench is a containerized, cloud-native data science working environment that runs on OpenShift
```

> Workbenches run as pods and often contain common data science tools such as JupyterLab, libraries such as Tensorflow, or ready-to-use GPU capabilities.

Reviewer: ```suggestion
Workbenches run as openshift pods and  contain commonly used data science tools such as JupyterLab, libraries such as Tensorflow, or ready-to-use GPU capabilities.
```

RHODS provides several preconfigured container images for creating workbenches.

[NOTE]
====
To use the RHODS GPU capabilities, the NVIDIA GPU operator must be installed.
====

Workbench Images::
Container images to be used as workbench containers.
RHODS includes a number of images ready to use for different common working environments.
Each image contains a different set of libraries and versions.
`Pytorch`, `TensorFlow`, `Minimal Python`, or `Standard Data Science` are examples of these images.

[NOTE]
====

> You can create custom workbench images.

Reviewer: Move this into the actual definition text instead of a NOTE

====

Persistent Storage::
You can mount different storage types into a workspace to persist your work.
For example, you can use persistent volume claims.

Data connections::
A data connection is a map of configuration values for connecting workbenches to external cloud storage providers, such as AWS S3.
Data connections inject the configuration values as environment variables in the associated workbench.

> Pipelines::

Reviewer: Use Data Science Pipelines to distinguish between regular pipelines

A workflow that executes scripts or Jupyter notebooks in an automated fashion.
Pipelines are useful to standardize the `data gathering - data cleaning - model training - model evaluation - model upload` cycle.
Typically, a pipeline ends after the trained model has been uploaded to an S3 bucket or similar cloud storage.

> [NOTE]

Reviewer: Move this into the actual definition text and delete the NOTE

====
To use pipelines in RHODS, the Red{nbsp}Hat OpenShift Pipelines operator must be installed.
====

Model servers::
A model server can read an exported model file and serve the model in a standard API.
Model servers require a data connection to download the model files from the cloud storage.

> RHODS is based on the _Open Data Hub_ upstream project.

Reviewer: Bring out more clearly the differences between RHODS and ODH. Maybe use a table format and fill in some points. We can ask BU to add other points that we missed.

Author: Added a table. What I added seems quite evident. The BU folks might want to take a look and add more specific differences.

https://opendatahub.io/[Open Data Hub] is an open source, community-driven AI platform, designed for the hybrid cloud, and built on popular AI open source tools.
RHODS is a subset of Open Data Hub, delivered both as a cloud service or self-managed.

> RHODS can be optionally delivered with additional independent software vendor (ISV) features.

Reviewer: Add a separate sub-section on this page and talk abouta typical data science project workflow, that is address this point in the design doc "Some concepts might be the steps in the model lifecycle like data prep, model building, training, model serving, inferencing and model monitoring." You can use definition lists to briefly explain data preparation, model building, training. serving, inferencing and monitoring and map it to corresponding architectural components. A diagram would be very nice here.

Check out these slides - [https://docs.google.com/presentation/d/1uJi_nmgTriDex1jVxUg-f2gIHElGfR1rLPJSLHmfJ7U/edit#slide=id.g11edfb70c9f_0_176](https://docs.google.com/presentation/d/1uJi_nmgTriDex1jVxUg-f2gIHElGfR1rLPJSLHmfJ7U/edit?authuser=0#slide=id.g11edfb70c9f_0_176)

Author: Done
